src/OrbitQt/orbittreeview.cpp

```diff
@@ -153,16 +153,20 @@ void OrbitTreeView::Refresh(RefreshMode refresh_mode) {
   }
 
   // Re-select previous selection
-  QItemSelectionModel* selection = selectionModel();
+  QItemSelectionModel* selection_model = selectionModel();
   QModelIndex index;
   // Don't re-trigger row selection callback when re-selecting.
   is_internal_refresh_ = true;
   std::vector<int> visible_selected_indices = model_->GetVisibleSelectedIndices();
-  selection->clearSelection();
+  selection_model->clearSelection();
+
+  QItemSelection selection{};
   for (int row : visible_selected_indices) {
     index = model_->CreateIndex(row, 0);
-    selection->select(index, QItemSelectionModel::Select | QItemSelectionModel::Rows);
+    selection.select(index, index);
   }
+  selection_model->select(selection, QItemSelectionModel::Select | QItemSelectionModel::Rows);
```

Comment:
Could you add the `QItemSelectionModel::Clear` flag here and remove the `clearSelection` call above? This would allow Qt to treat the selection change as an atomic operation and it might allow it to do some magic on how the current element is chosen. (Not sure if the algorithm actually does that though.)

Reply:
(And if doing so you probably don't need the `selection_model` variable anymore since there is only one call to `selectionModel()`

Reply:
done